### PR TITLE
Add more coverage of file_io methods in gcs_smoke tests.

### PR DESCRIPTION
**Pull request by sumitgouthaman**
Add more coverage of file_io methods in gcs_smoke tests.

Also adds some missing assertions in existing test cases.

### CODE REVIEW

File: tensorflow/tools/gcs_test/python/gcs_smoke.py [hunk 1]
```diff
@@ -83,22 +129,56 @@ def create_object_test():
   print("Creating dir %s." % dir_name)
   file_io.create_dir(dir_name)
 
-  # Create a file in this directory.
-  file_name = "%s/test_file.txt" % dir_name
-  print("Creating file %s." % file_name)
-  file_io.write_string_to_file(file_name, "test file creation.")
+  num_files = 5
+  # Create a files of 2 different patterns in this directory.
```
Comment (rinugun) [inline tensorflow/tools/gcs_test/python/gcs_smoke.py]: Typo: "create a files"

**Review by rinugun — APPROVED** [review]

File: tensorflow/tools/gcs_test/python/gcs_smoke.py (hunk 1 above)
Comment (sumitgouthaman) [inline tensorflow/tools/gcs_test/python/gcs_smoke.py]: Fixed

File: tensorflow/tools/gcs_test/python/gcs_smoke.py [hunk 2]
```diff
@@ -52,29 +52,75 @@ def create_examples(num_examples, input_mean):
 def create_dir_test():
   """Verifies file_io directory handling methods ."""
 
+  # Test directory creation
```
Comment (caisq) [inline tensorflow/tools/gcs_test/python/gcs_smoke.py]: nit: end comments with period.

File: tensorflow/tools/gcs_test/python/gcs_smoke.py [hunk 3]
```diff
@@ -52,29 +52,75 @@ def create_examples(num_examples, input_mean):
...
   elapsed = int(round(time.time() * 1000)) - starttime
   print("Created directory in: %d milliseconds" % elapsed)
+
   # Check that the directory exists.
   dir_exists = file_io.is_directory(dir_name)
+  assert dir_exists
   print("%s directory exists: %s" % (dir_name, dir_exists))
 
+  # Test recursive directory creation
+  starttime = int(round(time.time() * 1000))
```
Comment (caisq) [inline tensorflow/tools/gcs_test/python/gcs_smoke.py]: rename to starttime_ms to be explicit about the time unit?

File: tensorflow/tools/gcs_test/python/gcs_smoke.py [hunk 4]
```diff
@@ -52,29 +52,75 @@ def create_examples(num_examples, input_mean):
...
+  recursive_dir_name = "%s/%s/%s" % (dir_name,
+                                     "nested_dir1",
+                                     "nested_dir2")
+  print("Creating recursive dir %s" % recursive_dir_name)
+  file_io.recursive_create_dir(recursive_dir_name)
+  elapsed = int(round(time.time() * 1000)) - starttime
```
Comment (caisq) [inline tensorflow/tools/gcs_test/python/gcs_smoke.py]: same as above. rename to elapsed_ms to be explicit about time unit. same below.

File: tensorflow/tools/gcs_test/python/gcs_smoke.py [hunk 5]
```diff
@@ -52,29 +52,75 @@ def create_examples(num_examples, input_mean):
...
+  print("Created directory recursively in: %d milliseconds" % elapsed)
+
+  # Check that the directory exists.
+  recursive_dir_exists = file_io.is_directory(recursive_dir_name)
+  assert recursive_dir_exists
+  print("%s directory exists: %s" % (recursive_dir_name, recursive_dir_exists))
+
   # List contents of just created directory.
```
Comment (caisq) [inline tensorflow/tools/gcs_test/python/gcs_smoke.py]: Update comment to "Create some contents in the just created directory and list the contents."

File: tensorflow/tools/gcs_test/python/gcs_smoke.py [hunk 6]
```diff
@@ -52,29 +52,75 @@ def create_examples(num_examples, input_mean):
...
-  print(file_io.list_directory(dir_name))
+  directory_contents = file_io.list_directory(dir_name)
+  print(directory_contents)
   elapsed = int(round(time.time() * 1000)) - starttime
   print("Listed directory %s in %s milliseconds" % (dir_name, elapsed))
+  assert set(directory_contents) == set(files_to_create + ["nested_dir1/"])
+
+  # Test directory renaming
+  dir_to_rename = "%s/old_dir" % dir_name
+  new_dir_name = "%s/new_dir" %dir_name
```
Comment (caisq) [inline tensorflow/tools/gcs_test/python/gcs_smoke.py]: nit: missing space after the percent character.

**Review by caisq — CHANGES_REQUESTED** [review]

File: tensorflow/tools/gcs_test/python/gcs_smoke.py (hunk 2 above)
Comment (sumitgouthaman) [inline tensorflow/tools/gcs_test/python/gcs_smoke.py]: Fixed

File: tensorflow/tools/gcs_test/python/gcs_smoke.py (hunk 3 above)
Comment (sumitgouthaman) [inline tensorflow/tools/gcs_test/python/gcs_smoke.py]: fixed

File: tensorflow/tools/gcs_test/python/gcs_smoke.py (hunk 4 above)
Comment (sumitgouthaman) [inline tensorflow/tools/gcs_test/python/gcs_smoke.py]: fixed

File: tensorflow/tools/gcs_test/python/gcs_smoke.py (hunk 5 above)
Comment (sumitgouthaman) [inline tensorflow/tools/gcs_test/python/gcs_smoke.py]: fixed

File: tensorflow/tools/gcs_test/python/gcs_smoke.py (hunk 6 above)
Comment (sumitgouthaman) [inline tensorflow/tools/gcs_test/python/gcs_smoke.py]: fixed

**Review by caisq — APPROVED** [review]

**Pull request merged**